lib/rubocop/cop/performance/detect.rb

```diff
@@ -31,23 +33,34 @@ class Detect < Base
               '`%<first_method>s.%<second_method>s`.'
         REVERSE_MSG = 'Use `reverse.%<prefer>s` instead of ' \
                       '`%<first_method>s.%<second_method>s`.'
+        INDEX_MSG = 'Use `%<prefer>s` instead of ' \
+              '`%<first_method>s[%<index>i]`.'
+        INDEX_REVERSE_MSG = 'Use `reverse.%<prefer>s` instead of ' \
+              '`%<first_method>s[%<index>i]`.'
 
         def_node_matcher :detect_candidate?, <<~PATTERN
           {
             (send $(block (send _ {:select :find_all :filter}) ...) ${:first :last} $...)
+            (send $(block (send _ {:select :find_all :filter}) ...) $:[] (int ${0 -1}))
```

Comment:
It should also work with `[].filter(&:odd?).first`, right?

Reply:
Good catch, there were already tests for that with `first` and `last` but now it works with the array indices too.

Reply:
Thanks a lot for that patch :) Not gonna lie I had the PR coming as well but you got there first :+1: